lib/logstash/outputs/timetree/timetree.rb

```diff
@@ -0,0 +1,102 @@
+require 'neo4j'
+require 'logstash/outputs/timetree/model'
+
+module Neo4jrb
+  class TimeTree
+
+    attr_reader :root
+
+    def initialize(session)
+      @session = session
+      @root    = ::TimeTree::Root.first || create_node(::TimeTree::Root)
+    end
+
+    def add_event(ts, event)
+      Neo4j::Transaction.run do
+        start = refresh_tree(ts)
+        append(start, event)
+      end
+    end
+
+    def events_between(start_time, end_time)
+      raise Exception.new("Not implemented yet!")
+    end
+
+    def events_from(start_time)
+      raise Exception.new("Not implemented yet!")
+    end
+
+    def events_at(start_time)
+      query = <<-QUERY
+            MATCH (a:`TimeTree::Year`{value:#{start_time.year}})-[:`child`]->(b:`TimeTree::Month`{value:#{start_time.month}})-[:`child`]->(d:`TimeTree::Day`{value:#{start_time.day}})
+            WITH d MATCH (e:`TimeTree::Event`)<-[:`child`]-(d)
+            RETURN e
+        QUERY
```

Comment:
I would decouple the building of the query from the query execution, this make it easier to unit test.

```ruby
def build_query(start_time)
query = <<-QUERY
            MATCH (a:`TimeTree::Year`{value:#{start_time.year}})-[:`child`]->(b:`TimeTree::Month`{value:#{start_time.month}})-[:`child`]->(d:`TimeTree::Day`{value:#{start_time.day}})
            WITH d MATCH (e:`TimeTree::Event`)<-[:`child`]-(d)
            RETURN e
        QUERY
end
```